### PR TITLE
Draft get/setproperty overloading

### PR DESCRIPTION
**Pull request by simonbyrne**
Based on ideas from #245.

Which of the tests are a good benchmark?

### CODE REVIEW

File: test/DGmethods/compressible_Navier_Stokes/mms_bc.jl [hunk 1]
```diff
@@ -57,34 +57,66 @@ end
...
+
+function Base.getproperty(state::State, sym::Symbol)
+  arr = getfield(state,:arr)
+  if sym == :ρ
+    @inbounds arr[_ρ]
+  elseif sym == :ρ𝐮
+    @inbounds SVector(arr[_U], arr[_V], arr[_W])
+  elseif sym == :𝐮
+    @inbounds SVector(arr[_U], arr[_V], arr[_W]) ./ arr[_ρ]
+  elseif sym == :E
```
Comment (tapios) [inline test/DGmethods/compressible_Navier_Stokes/mms_bc.jl]: I suggest we use ``\rho e`` for ``E`` for clarity and consistency with the rest. Similarly for ``\rho q`` for humidities per unit volume etc.

**Comment by jkozdon** [comment]
In terms of performance, `mms_bc` is not great due to the heavy use of `sin` and `cos` in the manufactured solution. That said, functionality wise it has much of the complication including the fact that the `velocities!` only receives a partial state instead of the full state. Unfortunately, it gets the first 4 values. This could be made more complicated by changing
```julia
const _ρ, _U, _V, _W, _E = 1:_nstate
```
to
```julia
const _E, _ρ, _U, _V, _W = 1:_nstate
```

In general, for performance reason, in the future will likely need a number of the functions to get only the partial set of components from `Q`, `QV`, and `aux` that they need instead of the full set. Right now only the `gradient_transform!` callback has this functionality.

The reason this might be needed is that loading data out of the global arrays we don't need will degrade performance (unless we can do it in a way that the on the compiler to optimize the erroneous loads away?)

**Comment by jkozdon** [comment]
Some of the stuff @trontrytel @asraero and @smarras79 probably have more complication as well.

`isentropic_vortex_standalone` might be our best performing kernel since it's the simplest right now. Though I haven't done a thorough profiling of the code.

**Comment by simonbyrne** [comment]
Fixed by #266.

**Pull request closed without merging**